### PR TITLE
adding types

### PR DESCRIPTION
**Pull request by elopezanaya**
# PR Details

### Description

_Include a description of the problem to be solved_

## Solution Proposed

_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -231,19 +232,19 @@ class OmnichannelChatSDK {
...
-            return this.AMSClient;
+            case AMSClientLoadStates.ERROR:
+            case AMSClientLoadStates.NOT_LOADED:
+                await this.loadAmsClient();
+                return this.AMSClient;
 
-        default:
-            return null;
+            default:
+                return null;
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:247]: These are getting an eslint error
> Expected indentation of 8 spaces but found 12.eslint[indent](https://eslint.org/docs/rules/indent)

File: src/types/adapter.d.ts [hunk 2]
```diff
@@ -47,4 +47,18 @@ export type AdapterCreator = (
 	displayName: string,
 	chatClient: ChatClient,
 	logger: ACSAdapterLogger,
-	featureOption: IAdapterInitFeatureOption) => IChatAdapter;
\ No newline at end of file
+	featureOption: IAdapterInitFeatureOption) => IChatAdapter;
+
+export type OmnichannelGenericResponse = {
+	data : {} | undefined;
```
Comment (matan-hershberg-nuance) [inline src/types/adapter.d.ts:53]: When defining somthing as `{}` it can also mean that it could be a number, null, string. It behaves more like `any`.
But if we know it's an object with properties like `{ key: "value" }` it's better to use a record
```suggestion
	data? : Record<string, unknown>;
```

File: src/OmnichannelChatSDK.ts [hunk 3]
```diff
@@ -1791,7 +1835,7 @@ class OmnichannelChatSDK {
         return this.callingOption.toString() !== CallingOptionsOptionSetNumber.NoCalling.toString();
     }
 
-    public async getVoiceVideoCalling(params: any = {}): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
+    public async getVoiceVideoCalling(params: any = {}): Promise<VoiceVideoCallingProxy | undefined> { // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:1838]:
```suggestion
    public async getVoiceVideoCalling(params: any = {}): Promise<VoiceVideoCallingProxy | undefined> {
```

File: src/OmnichannelChatSDK.ts [hunk 4]
```diff
@@ -2004,10 +2090,26 @@ class OmnichannelChatSDK {
...
+            };
+
+            return executionResult;
         } catch (e) {
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            reportError("GetAgentAvailabilityFailed", (e as any).message as string);
+
+            const exceptionDetails: ChatSDKExceptionDetails = {
+                response: "GetAgentAvailabilityFailed",
+                message: (e as Error)?.message
```
Comment (matan-hershberg-nuance) [inline src/OmnichannelChatSDK.ts:2104]:
```suggestion
                message: e instanceof Error ? e.message : String(e)
```

File: src/types/adapter.d.ts (hunk 2 above)
Comment (elopezanaya) [reply src/types/adapter.d.ts:53]: the idea is to be a generic wrapper
